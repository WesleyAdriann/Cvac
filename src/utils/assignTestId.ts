import { Platform } from 'react-native'

interface ITestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}

type TComponentType =
  | 'View'
  | 'ScrollView'
  | 'TextInput'
  | 'Text'
  | 'SafeAreaView'
  | 'Pressable'
  | 'Modal'

export const assignTestId = (
  componentType: TComponentType,
  componentId: string
): ITestID => {
  const id = `${componentType}_${componentId}`
  return Platform.OS === 'android'
    ? {
        testID: id,
        accessible: true,
        accessibilityLabel: id
      }
    : {
        testID: id,
        accessible: false
      }
}
export const mockTestID = (
  componentType: TComponentType,
  componentId: string
): ITestID => assignTestId(componentType, componentId)
