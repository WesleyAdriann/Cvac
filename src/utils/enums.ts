export enum ECalendarsName {
  'kid' = 'Crian√ßa',
  'elder' = 'Idoso',
  'adult' = 'Adulto',
  'teen' = 'Adolescente'
}
