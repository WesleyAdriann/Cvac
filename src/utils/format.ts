export const formatDate = (value: string) => {
  const clearDate = value.replace(/\D/g, '')

  if (clearDate.length === 8) return value

  const day = clearDate.slice(0, 2)
  const month = clearDate.slice(2, 4)
  const year = clearDate.slice(4, 8)

  if (!month) return day
  if (!year) return `${day}/${month}`
  return `${day}/${month}/${year}`
}
