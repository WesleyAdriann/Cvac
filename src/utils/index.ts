export * from './assignTestId'
export * from './withProviders'
export * from './format'
export * from './logs'
