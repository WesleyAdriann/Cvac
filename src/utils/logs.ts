export const log = (TAG: string, ...args: any[]) => {
  if (__DEV__) console.log(`[${TAG}] `, ...args)
}
