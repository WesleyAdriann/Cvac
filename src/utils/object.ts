export const isEmpty = (obj: object | null) => {
  if (!obj) return true
  // eslint-disable-next-line no-unreachable-loop
  for (const key in obj) return false
  return true
}
