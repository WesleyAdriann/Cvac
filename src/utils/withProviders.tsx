import React from 'react'
import { Provider as PaperProvider } from 'react-native-paper'
import { ThemeProvider } from 'styled-components'

import { theme, paperTheme } from '../tokens'

export const TestsProviders: React.FC = ({ children }) => (
  <PaperProvider theme={paperTheme} >
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </PaperProvider>
)
