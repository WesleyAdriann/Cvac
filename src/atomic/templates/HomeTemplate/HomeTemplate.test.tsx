
import React from 'react'
import { render, cleanup } from '@testing-library/react-native'

import { AllTheProviders as wrapper, mockTestID } from '~/utils'

import { HomeTemplate, IHomeTemplate } from './index'

describe('Templates/HomeTemplate', () => {
  const props: IHomeTemplate = {
    testID: 'HomeTemplate',
    onPressAuthItem: jest.fn(),
    onPressMenuItem: jest.fn(),
    authIsLoading: false
  }

  const elContainer = mockTestID('SafeAreaView', props.testID!).testID

  const renderComponent = () => render(<HomeTemplate {...props} />, { wrapper })
  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  it(`should render component #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
