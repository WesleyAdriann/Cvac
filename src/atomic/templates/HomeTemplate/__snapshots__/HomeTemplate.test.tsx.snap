// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/HomeTemplate should render snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    accessible={false}
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    testID="SafeAreaView_HomeTemplate"
  >
    <RCTScrollView
      accessible={false}
      contentContainerStyle={
        Object {
          "flexGrow": 1,
          "padding": 16,
        }
      }
      flex={1}
      paddingStyle={16}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
        ]
      }
      testID="ScrollView_Flex"
    >
      <View>
        <View
          accessible={false}
          alignItems="center"
          flex={1}
          justifyContent="center"
          marginStyle={16}
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 16,
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 16,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          testID="View_Flex"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/logotipo/logotipo.png",
              }
            }
          />
        </View>
        <View
          accessible={false}
          flex={2}
          justifyContent="center"
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 2,
                "flexShrink": 1,
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          testID="View_Flex"
        >
          <View
            accessible={false}
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
            testID="View_Flex"
          >
            <View
              accessible={false}
              flexDirection="row"
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                ]
              }
              testID="View_Flex"
            >
              <View
                collapsable={false}
                flex={1}
                marginStyle="0 16px 0 0"
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "#00C9D2",
                    "borderColor": "#00C9D2",
                    "borderRadius": 4,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "elevation": 0,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 16,
                    "marginTop": 0,
                    "minWidth": 64,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 4,
                      },
                    ]
                  }
                  testID="Pressable_Button"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        Object {
                          "paddingVertical": 6,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#424242",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "letterSpacing": 1,
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            undefined,
                            Object {
                              "textTransform": "uppercase",
                            },
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "500",
                            },
                            Object {
                              "fontFamily": "System",
                              "fontWeight": "500",
                            },
                            Object {
                              "color": "#fff",
                              "fontSize": 18,
                            },
                          ],
                        ]
                      }
                    >
                      <Text
                        mode="contained"
                        style={
                          Array [
                            Object {
                              "color": "#424242",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "color": "#fff",
                                "fontSize": 18,
                                "fontWeight": "bold",
                              },
                            ],
                          ]
                        }
                      >
                        login
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                flex={1}
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#00A3C9",
                    "borderRadius": 4,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "elevation": 0,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "minWidth": 64,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 4,
                      },
                    ]
                  }
                  testID="Pressable_Button"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        Object {
                          "paddingVertical": 6,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#424242",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "letterSpacing": 1,
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            undefined,
                            Object {
                              "textTransform": "uppercase",
                            },
                            Object {
                              "color": "#00C9D2",
                              "fontFamily": "System",
                              "fontWeight": "500",
                            },
                            Object {
                              "fontFamily": "System",
                              "fontWeight": "500",
                            },
                            Object {
                              "color": "#00C9D2",
                              "fontSize": 18,
                            },
                          ],
                        ]
                      }
                    >
                      <Text
                        mode="outlined"
                        style={
                          Array [
                            Object {
                              "color": "#424242",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "color": "#00C9D2",
                                "fontSize": 18,
                                "fontWeight": "normal",
                              },
                            ],
                          ]
                        }
                      >
                        cadastrar
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            accessible={false}
            marginStyle="16px 0 0"
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 16,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
            testID="View_Flex"
          >
            <View
              accessible={false}
              flexDirection="row"
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                ]
              }
              testID="View_Flex"
            >
              <View
                accessible={false}
                collapsable={false}
                focusable={true}
                gap={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#00A3C9",
                      "borderRadius": 2,
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "height": 351,
                      "justifyContent": "center",
                      "marginRight": 16,
                      "paddingBottom": 14,
                      "paddingLeft": 14,
                      "paddingRight": 14,
                      "paddingTop": 14,
                    },
                  ]
                }
                testID="Pressable_HomeItem"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#00C9D2",
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "fontSize": 28,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰃭
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#424242",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "color": "#00C9D2",
                          "fontSize": 16,
                          "textAlign": "center",
                          "textTransform": "uppercase",
                        },
                      ],
                    ]
                  }
                >
                  calendarios de vacinação
                </Text>
              </View>
              <View
                accessible={false}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#00A3C9",
                      "borderRadius": 2,
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "height": 351,
                      "justifyContent": "center",
                      "marginRight": 0,
                      "paddingBottom": 14,
                      "paddingLeft": 14,
                      "paddingRight": 14,
                      "paddingTop": 14,
                    },
                  ]
                }
                testID="Pressable_HomeItem"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#00C9D2",
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "fontSize": 28,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰸎
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#424242",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "color": "#00C9D2",
                          "fontSize": 16,
                          "textAlign": "center",
                          "textTransform": "uppercase",
                        },
                      ],
                    ]
                  }
                >
                  carteira de vacinação
                </Text>
              </View>
            </View>
            <View
              accessible={false}
              flexDirection="row"
              marginStyle="16px 0 0"
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 16,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                ]
              }
              testID="View_Flex"
            >
              <View
                accessible={false}
                collapsable={false}
                focusable={true}
                gap={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#00A3C9",
                      "borderRadius": 2,
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "height": 351,
                      "justifyContent": "center",
                      "marginRight": 16,
                      "paddingBottom": 14,
                      "paddingLeft": 14,
                      "paddingRight": 14,
                      "paddingTop": 14,
                    },
                  ]
                }
                testID="Pressable_HomeItem"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#00C9D2",
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "fontSize": 28,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰂚
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#424242",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "color": "#00C9D2",
                          "fontSize": 16,
                          "textAlign": "center",
                          "textTransform": "uppercase",
                        },
                      ],
                    ]
                  }
                >
                  lembretes de vacina
                </Text>
              </View>
              <View
                accessible={false}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#00A3C9",
                      "borderRadius": 2,
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "height": 351,
                      "justifyContent": "center",
                      "marginRight": 0,
                      "paddingBottom": 14,
                      "paddingLeft": 14,
                      "paddingRight": 14,
                      "paddingTop": 14,
                    },
                  ]
                }
                testID="Pressable_HomeItem"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#00C9D2",
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "fontSize": 28,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰍎
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#424242",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "color": "#00C9D2",
                          "fontSize": 16,
                          "textAlign": "center",
                          "textTransform": "uppercase",
                        },
                      ],
                    ]
                  }
                >
                  locais de vacinação
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
